File: src/main/java/at/ac/tuwien/trustcps/grid/IntegerDistance.java
package at.ac.tuwien.trustcps.grid;

import eu.quanticol.moonlight.formula.DistanceDomain;

public class IntegerDistance implements DistanceDomain<Integer> {
    public IntegerDistance() { /* TODO document why this constructor is empty */ }

    public Integer zero() {
        return 0;
    }

    public Integer infinity() {
        return 1 / 0;
    }

    public boolean lessOrEqual(Integer x, Integer y) {
        return x < y || this.equalTo(x, y);
    }

    public Integer sum(Integer x, Integer y) {
        return x + y;
    }

    public boolean equalTo(Integer x, Integer y) {
        return Math.abs(x - y) < 1.0E-12D;
    }

    public boolean less(Integer x, Integer y) {
        return x < y;
    }
}

File: src/main/kotlin/space/IntegerDistance.kt
package space

import eu.quanticol.moonlight.formula.DistanceDomain

class IntegerDistance : DistanceDomain<Int> {
    override fun zero(): Int {
        return 0
    }

    override fun infinity(): Int {
        return 1 / 0
    }

    override fun lessOrEqual(x: Int, y: Int): Boolean {
        return x < y || equalTo(x, y)
    }

    override fun sum(x: Int, y: Int): Int {
        return x + y
    }

    override fun equalTo(x: Int, y: Int): Boolean {
        return Math.abs(x - y) < 1.0E-12
    }

    override fun less(x: Int, y: Int): Boolean {
        return x < y
    }
}
